src/data/actions/stafferInfo.js

```diff
@@ -83,14 +84,20 @@ export function editStafferInfo(data) {
   return { type: EDIT_STAFFER_INFO, data };
 }
 
+export function editStafferInfoFinish() {
+  return { type: EDIT_STAFFER_INFO_FINISH };
+}
+
 export function editStaffer(stafferData) {
   return (dispatch) => {
     dispatch(editStafferInfo(stafferData));
     // Send edit course PATCH
     return DiscoveryDataApiService.editStaffer(stafferData)
       .then((response) => {
         const staffer = response.data;
-        dispatch(editStafferInfoSuccess(staffer));
+        dispatch(editStafferInfoSuccess(staffer))
+          // Used to reset the wasEditSuccessful flag after a successful edit
+          .then(dispatch(editStafferInfoFinish()));
```

Comment:
`editStafferInfoFinish` is resetting the `wasEditSuccessful` flag in state after `editStafferInfoSuccess`, so that you don't get redirected immediately whenever you try to go any instructor page again

Reply:
Alternatively, I could redirect here after every successful edit if there is a referrer passed in to this function. Does that feel cleaner to you?

Reply:
In my mind, which one is nicer depends on if we ever plan for people to use the instructor page by itself, or if they only use it coming from somewhere else

Reply:
I imagine it's always coming from somewhere else? But I'm loathe to commit to that.

Reply:
Updated to do it in a better™️ way